eradiate/radprops/rad_profile.py

```diff
@@ -345,76 +368,29 @@ def _validator_values(instance, attribute, value):
                 f"the same length"
             )
 
-    _wavelength = pinttr.ib(
-        default=None, units=ucc.deferred("wavelength"), init=False, repr=True
-    )
-
-    def __attrs_post_init__(self):
-        self.update()
-
-    def update(self):
-        mode = eradiate.mode()
-        if not mode.is_monochromatic():
-            raise ModeError(f"unsupported mode {mode.id}")
-        else:
-            wavelength = mode.wavelength
-            self._wavelength = wavelength
-
-    @property
-    def albedo(self):
-        """
-        Return albedo.
-
-        Returns → :class:`pint.Quantity`:
-            Profile albedo.
-        """
+    def albedo(self, spectral_ctx=None):
         return self.albedo_values
 
-    @property
-    def sigma_t(self):
-        """
-        Return extinction coefficient.
-
-        Returns → :class:`pint.Quantity`:
-            Profile extinction coefficient.
-        """
+    def sigma_t(self, spectral_ctx=None):
         return self.sigma_t_values
 
-    @property
-    def sigma_a(self):
-        """
-        Return absorption coefficient.
+    def sigma_a(self, spectral_ctx=None):
+        return self.sigma_t(spectral_ctx) * (1.0 - self.albedo(spectral_ctx))
 
-        Returns → :class:`pint.Quantity`:
-            Profile absorption coefficient.
-        """
-        return self.sigma_t * (1.0 - self.albedo)
-
-    @property
-    def sigma_s(self):
-        """
-        Return scattering coefficient.
-
-        Returns → :class:`pint.Quantity`:
-            Profile scattering coefficient.
-        """
-        return self.sigma_t * self.albedo
+    def sigma_s(self, spectral_ctx=None):
+        return self.sigma_t(spectral_ctx) * self.albedo(spectral_ctx)
 
-    def to_dataset(self):
-        """
-        Return a dataset that holds the radiative properties of the
-        corresponding atmospheric radiative properties profile.
+    def to_dataset(self, spectral_ctx=None):
+        if eradiate.mode().is_monochromatic():
+            return make_dataset(
+                wavelength=spectral_ctx.wavelength,
```

Comment:
I'm currently working on this part of the code and it will need refactoring after the spectral loop stuff. Not quite sure about the relevance of a `spectral_ctx` arg, I'd actually be more in favour of a generic spectral coordinate arg. We'll see about that.